### PR TITLE
feat: Add "quit" command to `rustlings watch`

### PR DESCRIPTION
**Pull request by abusch**
closes: #842

Took a crack at adding a "quit" command to the `rustlings watch` prompt, as at the moment the only way to exit that shell is to finish all the exercises in one go.

The implementation is a bit clunky... Ideally I would have another channel to signal the intention to quit from the thread spawned by `spawn_watch_shell()`, and the `watch` function would `select` on both receivers, but I don't think there's a way to do that with `std` channels. We can't move to `crossbeam` channels either as the `notify` crate has a hard dependency on `std` channels :( (this is fixed in the upcoming version 5, but it hasn't been released yet...).

I probably should have waited for some discussion on the issue, but it seemed easy enough to give it a shot anyway... Hope this can be useful.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -217,65 +218,76 @@ fn main() {
...
+                        "If you noticed any issues, please don't hesitate to report them to our repo."
+                    );
+                    println!("You can also contribute your own exercises to help the greater community!");
+                    println!();
+                    println!("Before reporting an issue or contributing, please read our guidelines:");
+                    println!("https://github.com/rust-lang/rustlings/blob/main/CONTRIBUTING.md");
+                }
+                Ok(WatchStatus::Unfinished) => {
+                    println!("We hope you're enjoying learning about Rust!");
+                    println!("If you want to continue working on the exercises at a later point, you can simply run `rustlings watch` again");
```
Comment (abusch) [inline src/main.rs:267]: More than happy to get better copy here :)

**Comment by diannasoriel** [comment]
@abusch Oops, totally accidentally superseded part of this PR with mine (#846). Would you be up for converting this PR into one that only adds the quit-after-finish functionality? That part looks good, but it'll lead to merge conflicts if we both add the quit command.

**Comment by abusch** [comment]
@diannasoriel ha, no worries :)
I see your PR just got merged so I'll rebase mine and see how it goes.

**Comment by diannasoriel** [comment]
@abusch Thank you! Looks good.

**Comment by diannasoriel** [comment]
@all-contributors please add @abusch for code

**Pull request merged**